### PR TITLE
bug: track added install paths

### PR DESCRIPTION
**Pull request by jdxcode**
This changes the logic for `rtx activate` with and without direnv fairly significantly.

Before, when loading PATH, rtx would strip anything that had the prefix of `~/.local/share/rtx/installs`, that way it would remove installs that were added in any rtx run. However there may be paths that were added outside of rtx which should be kept. There was also a direnv bug that we didn't perform this stripping on DIRENV_DIFF, so when moving from a direnv directory that used rtx to one that didn't, direnv would add rtx paths back when it shouldn't.

This all makes things smarter and rtx will track all of the install paths that were used the last time `hook-env` was run. It will then remove each of those paths from both PATh and DIRENV_DIFF one by one to get back to the "pristine" state before rtx would've run.

This code is all quite complex, but we're slowly getting more test cases in place and I think we're resolving these edge cases.

### CODE REVIEW

**Comment by jdxcode** [comment]
# [Codecov](https://codecov.io/gh/jdxcode/rtx/pull/89?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Jeff+Dickey) Report
Base: **81.47**% // Head: **81.04**% // Decreases project coverage by **`-0.44%`** :warning:
> Coverage data is based on head [(`cd1238a`)](https://codecov.io/gh/jdxcode/rtx/pull/89?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Jeff+Dickey) compared to base [(`6798ab1`)](https://codecov.io/gh/jdxcode/rtx/commit/6798ab1afc4d4d8b530ff0ffc7532b3deb6b8175?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Jeff+Dickey).
> Patch coverage: 55.88% of modified lines in pull request are covered.

<details><summary>Additional details and impacted files</summary>


```diff
@@            Coverage Diff             @@
##             main      #89      +/-   ##
==========================================
- Coverage   81.47%   81.04%   -0.44%     
==========================================
  Files          78       78              
  Lines        2375     2400      +25     
==========================================
+ Hits         1935     1945      +10     
- Misses        440      455      +15     
```


| [Impacted Files](https://codecov.io/gh/jdxcode/rtx/pull/89?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Jeff+Dickey) | Coverage Δ | |
|---|---|---|
| [src/direnv.rs](https://codecov.io/gh/jdxcode/rtx/pull/89?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Jeff+Dickey#diff-c3JjL2RpcmVudi5ycw==) | `79.16% <0.00%> (-20.84%)` | :arrow_down: |
| [src/cli/hook\_env.rs](https://codecov.io/gh/jdxcode/rtx/pull/89?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Jeff+Dickey#diff-c3JjL2NsaS9ob29rX2Vudi5ycw==) | `71.42% <66.66%> (-2.77%)` | :arrow_down: |
| [src/env.rs](https://codecov.io/gh/jdxcode/rtx/pull/89?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Jeff+Dickey#diff-c3JjL2Vudi5ycw==) | `68.18% <85.71%> (+2.42%)` | :arrow_up: |
| [src/env\_diff.rs](https://codecov.io/gh/jdxcode/rtx/pull/89?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Jeff+Dickey#diff-c3JjL2Vudl9kaWZmLnJz) | `87.83% <100.00%> (+0.16%)` | :arrow_up: |

Help us with your feedback. Take ten seconds to tell us [how you rate us](https://about.codecov.io/nps?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Jeff+Dickey). Have a feature suggestion? [Share it here.](https://app.codecov.io/gh/feedback/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Jeff+Dickey)

</details>

[:umbrella: View full report at Codecov](https://codecov.io/gh/jdxcode/rtx/pull/89?src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Jeff+Dickey).   
:loudspeaker: Do you have feedback about the report comment? [Let us know in this issue](https://about.codecov.io/codecov-pr-comment-feedback/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Jeff+Dickey).

**Comment by jdxcode** [comment]
this passes but it is causing `PATH` to keep growing after every `cd`. I'm going to write a test to prevent that from happening now and in the future.

**Comment by jdxcode** [comment]
ignore codecov here, E2E tests aren't tracked by coverage

File: src/snapshots/rtx__direnv__test__add_path_to_old_and_new.snap [hunk 1]
```diff
@@ -2,4 +2,4 @@
 source: src/direnv.rs
 expression: "diff.old.get(\"PATH\").unwrap()"
 ---
-/tmp:/foo:/bar:/old
+/tmp:/foo:/tmp:/bar:/old
```
Comment (jdx) [inline src/snapshots/rtx__direnv__test__add_path_to_old_and_new.snap:5]: this changed because we're not removing old paths anymore
